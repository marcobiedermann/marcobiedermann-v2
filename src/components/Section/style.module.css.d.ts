export const section: string;
export const sectionDark: string;
export const sectionFull: string;
export const sectionLight: string;
