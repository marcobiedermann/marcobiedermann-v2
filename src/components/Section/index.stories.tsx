/* eslint-disable import/no-extraneous-dependencies */

import { storiesOf } from '@storybook/react';
import React from 'react';
import Section from '.';

storiesOf('Section', module).add('default', () => <Section>Content</Section>);
