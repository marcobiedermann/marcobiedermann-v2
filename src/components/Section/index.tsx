import classNames from 'classnames';
import capitalize from 'lodash/capitalize';
import React from 'react';
import styles from './style.module.css';

type Appearance = 'dark' | 'light';

export interface SectionProps {
  appearance?: Appearance;
  className?: string;
  isFull?: boolean;
}

const Section: React.FC<SectionProps> = (props) => {
  const { appearance, children, className, isFull, ...otherProps } = props;

  return (
    <section
      className={classNames(className, styles.section, {
        [styles.sectionFull]: isFull,
        [styles[`section${capitalize(appearance)}` as 'sectionLight' | 'sectionDark']]: appearance,
      })}
      {...otherProps}
    >
      {children}
    </section>
  );
};

export default Section;
