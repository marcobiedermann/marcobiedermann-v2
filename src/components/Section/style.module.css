.section {
  align-items: center;
  display: flex;
  padding-bottom: 10em;
  padding-top: 10em;
}

.section--dark {
  /* background-color: #0a0a0a; */
  background-image: radial-gradient(#24252e, #1a1c21);
  color: #f8f9fb;
}

.section--full {
  min-height: 100vh;
}

.section--light {
  background-color: #eee;
}
