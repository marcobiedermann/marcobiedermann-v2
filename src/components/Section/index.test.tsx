/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Section from '.';

describe('Section component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Section />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render dark appearance correctly', () => {
    const tree = renderer.create(<Section appearance="dark" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render light appearance correctly', () => {
    const tree = renderer.create(<Section appearance="light" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render full mode correctly', () => {
    const tree = renderer.create(<Section isFull />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
