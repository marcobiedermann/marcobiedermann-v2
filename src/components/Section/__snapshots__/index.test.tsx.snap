// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section component should render correctly 1`] = `
<section
  className="section"
/>
`;

exports[`Section component should render dark appearance correctly 1`] = `
<section
  className="section sectionDark"
/>
`;

exports[`Section component should render full mode correctly 1`] = `
<section
  className="section sectionFull"
/>
`;

exports[`Section component should render light appearance correctly 1`] = `
<section
  className="section sectionLight"
/>
`;
